embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/EmbraceSpansService.kt

```diff
@@ -13,7 +13,7 @@ import io.opentelemetry.api.trace.Tracer
 internal class EmbraceSpansService(
     private val spansRepository: SpansRepository,
     private val currentSessionSpan: CurrentSessionSpan,
-    private val tracer: Tracer,
+    private val tracerSupplier: () -> Tracer,
```

Comment:
We're using this pattern a lot more frequently - I wonder whether it makes sense to create a typealias that makes this a bit more explicit? E.g. something like:

```
internal typealias Provider<T> = () -> T
```

Not required for this changeset

Reply:
Good idea! Provider is a bit Dagger-esque, and Supplier is a it Java-esque, so either one would work. Lets do that after.